app/main.py

```diff
@@ -1,2 +1,46 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.threshold = 0.67 * self.capacity
+        self.bucket = [[]] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        empty_slot = hash(key) % self.capacity
+        while True:
+            if not self.bucket[empty_slot]:
+                self.bucket[empty_slot] = [key, hash(key), value]
+                self.length += 1
+                break
+            if self.bucket[empty_slot][0] == key and (
+                    self.bucket[empty_slot][1] == hash(key)):
+                self.bucket[empty_slot][2] = value
+                break
+            empty_slot = (empty_slot + 1) % self.capacity
+        if self.length > self.threshold:
+            self.resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        empty_slot = hash(key) % self.capacity
+        while self.bucket[empty_slot]:
+            if key == self.bucket[empty_slot][0] and (
+                    self.bucket[empty_slot][1] == hash(key)):
+                return self.bucket[empty_slot][2]
+            empty_slot = (empty_slot + 1) % self.capacity
+        raise KeyError(key)
+
+    def __len__(self) -> int:
+        return self.length
+
+    def resize(self) -> None:
```

Comment:
add encapsulation for this method, it can only be used in this class

Reply:
Done